From 044cd3bd30a8867ee2afab577fd4e3ea0dbec046 Mon Sep 17 00:00:00 2001
From: Anson Huang <Anson.Huang@nxp.com>
Date: Mon, 5 Dec 2016 22:33:43 +0800
Subject: [PATCH 1306/5242] MLK-13565 ARM: dts: imx7ulp: enable debug uart for
 14x14 arm2 board

commit  d264b63475a2cc2b9e4341384ff45a7601123d8a from
https://source.codeaurora.org/external/imx/linux-imx.git

i.MX7ULP's lpuart4 is disabled in dtsi, need to enable it in
14x14 arm2 board for console.

Signed-off-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx7ulp-14x14-arm2.dts |   15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/imx7ulp-14x14-arm2.dts b/arch/arm/boot/dts/imx7ulp-14x14-arm2.dts
index 2921e6c..5278cb8 100644
--- a/arch/arm/boot/dts/imx7ulp-14x14-arm2.dts
+++ b/arch/arm/boot/dts/imx7ulp-14x14-arm2.dts
@@ -29,7 +29,7 @@
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_hog_1>;
 
-	imx7ulp-evk {
+	imx7ulp-14x14-arm2 {
 		pinctrl_hog_1: hoggrp-1 {
 			fsl,pins = <
 				IMX7ULP_PAD_PTC10__PTC10	0x30100
@@ -37,6 +37,13 @@
 			>;
 		};
 
+		pinctrl_lpuart4: lpuart4grp {
+			fsl,pins = <
+				IMX7ULP_PAD_PTC3__LPUART4_RX	0x400
+				IMX7ULP_PAD_PTC2__LPUART4_TX	0x400
+			>;
+		};
+
 		pinctrl_usdhc1: usdhc1grp {
 			fsl,pins = <
 				IMX7ULP_PAD_PTE3__SDHC1_CMD	0x843
@@ -50,6 +57,12 @@
 	};
 };
 
+&lpuart4 { /* console */
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_lpuart4>;
+	status = "okay";
+};
+
 &usdhc1 {
 	pinctrl-names = "default", "sleep";
 	pinctrl-0 = <&pinctrl_usdhc1>;
-- 
1.7.9.5

